library/alloc/src/collections/btree/node.rs

```diff
@@ -215,10 +225,10 @@ impl<K, V> Root<K, V> {
 
         let top = self.node.ptr;
 
+        let mut internal_node = unsafe { self.internal_node_as_mut() };
+        let internal_node = internal_node.as_internal_mut();
         self.node = unsafe {
-            BoxedNode::from_ptr(
-                self.node_as_mut().cast_unchecked::<marker::Internal>().first_edge().descend().node,
-            )
+            mem::replace(&mut internal_node.edges[0], MaybeUninit::uninit()).assume_init()
```

Comment:
This feels like we're (potentially) doing needless work -- this looks equivalent to `internal_node.edges[0].assume_init_read()` which avoids the spurious write to edges -- can we use that instead?

(I suspect in practice LLVM probably ignores writing uninitialized data to some region of memory, but not sure).

Reply:
Indeed and I did the logical equivalent ptr::read on the other side.